spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -268,17 +650,349 @@ public PublisherBuilder<IncomingMessage, OutgoingMessage> processMessages() {
+----
+@Incoming("in")
+@Outgoing("out") 
+Processor<I, O> method();
+----
+| Pre-Processing
+| None, Pre-Processing
+Post-Processing can be optionaly supported by implementations, however it requires a 1:1 mapping between the incoming message and the outgoing message.
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+ProcessorBuilder<Message<I>, Message<O>> method();
+----
+| Pre-Processing
+| None, Pre-Processing, Manual
+Post-Processing can be optionaly supported by implementations, however it requires a 1:1 mapping between the incoming message and the outgoing message.
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+ProcessorBuilder<I, O> method();
+----
+| Pre-Processing
+| None, Pre-Processing
+Post-Processing can be optionaly supported by implementations, however it requires a 1:1 mapping between the incoming message and the outgoing message.
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Publisher<Message<O>> method(Message<I> msg)
+----
+| Pre-Processing
+| None, Manual, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Publisher<O> method(I payload)
+----
+| Pre-Processing
+| None, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+PublisherBuilder<Message<O>> method(Message<I> msg)
+----
+| Pre-Processing
+| None, Manual, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+PublisherBuilder<O> method(I payload)
+----
+| Pre-Processing
+| None, Pre-Processing
+
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Message<O> method(Message<I> msg)
+----
+| Post-Processing
+| None, Manual, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+O method(I payload)
+----
+| Post-Processing
+| None, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+CompletionStage<Message<O>> 
+method(Message<I> msg)
+----
+| Post-Processing
+| None, Manual, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+CompletionStage<O> method(I payload)
+----
+| Post-Processing
+| None, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Publisher<Message<O>> method(Publisher<Message<I>> pub)
+----
+| Pre-Processing
+| None, Manual, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+PublisherBuilder<Message<O>> method(PublisherBuilder<Message<I>> pub)
+----
+| Pre-Processing
+| None, Manual, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Publisher<O> method(Publisher<I> pub)
+----
+| Pre-Processing
+| None, Pre-Processing
+
+|
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+PublisherBuilder<O> method(PublisherBuilder<I> pub)
+----
+| Pre-Processing
+| None, Pre-Processing
+
+|===
+
+Invalid acknowledgement policies must be detected and a `DeploymentException` raised when the application starts.
```

Comment:
...is deployed.        (It is possible to have a stopped but deployed application.)